src/state_machine.rs

```diff
@@ -277,15 +267,7 @@ pub enum Transition {
         details: SignedRelocateDetails,
     },
     // `JoiningPeer` state transitioning to `Adult`.
-    IntoAdult {
-        gen_pfx_info: GenesisPfxInfo,
-    },
-    // `Adult` state transition to `Elder`.
-    IntoElder {
-        old_pfx: Prefix<XorName>,
-    },
-    // `Elder` state transition to `Adult` as a result of demotion
-    Demote {
+    Approve {
```

Comment:
the comment says `JoiningPeer state transitioning to Adult`, maybe shall be `to Approved` ?
also, `Approve` to `IntoApproved` ?

Reply:
:+1 for the comment. Regarding `Approve` vs `IntoApproved` - I kind of like the transition variants to be verbs describing what is happening rather than what are we transitioning into. But this whole enum is going away in the next PR (or the one after) anyway, so I wouldn't worry about it too much.